aptos-move/aptos-intent/src/builder.rs

```diff
@@ -0,0 +1,473 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{ArgumentOperation, BatchArgument, BatchedFunctionCall, PreviousResult};
+use anyhow::{anyhow, bail, Result};
+use move_binary_format::{
+    access::ModuleAccess,
+    binary_views::BinaryIndexedView,
+    file_format::{Ability, AbilitySet, FunctionHandle, SignatureToken},
+    CompiledModule,
+};
+use move_core_types::{
+    identifier::{IdentStr, Identifier},
+    language_storage::{ModuleId, StructTag, TypeTag},
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{collections::BTreeMap, str::FromStr};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+/// Arguments for each function.
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum BatchArgumentType {
+    Raw,
+    Signer,
+    PreviousResult,
+}
+
+#[wasm_bindgen]
+/// Arguments for each function. Wasm bindgen only support C-style enum so use option to work around.
```

Comment:
What do we do if more than one is not `None`?

Reply:
This is a variant we need to maintain for our own implementation.